AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/QtPyScriptCanvasNodeInspector.py

```diff
@@ -0,0 +1,109 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+
+Object to house all the Qt Objects and behavior used in testing the script canvas Node Inspector.
+
+***Important Note: The QWidget hierarchy of the node inspector changes based on what node/variable is selected***
+"""
+
+from editor_python_test_tools.utils import TestHelper as helper
+from PySide2 import QtWidgets, QtTest, QtCore
+import pyside_utils
+from consts.scripting import (PROPERTY_EDITOR_QT, INITIAL_VALUE_SOURCE_QT)
+from consts.general import (WAIT_TIME_SEC_1)
+from enum import IntEnum
+
+class BooleanCheckBoxValues(IntEnum): # get a better name for this
+    Off = 0
+    On = 1
+
+class QtPyScriptCanvasNodeInspector:
+
+    def __init__(self, sc_editor):
+        self.node_inspector = sc_editor.sc_editor.findChild(QtWidgets.QDockWidget, "NodeInspector")
+
+
+    def change_variable_initial_value(self, variable_name: str, variable_type: str,  new_value) -> None:
+        """
+        Function for changing the initial value of a variable. This function will have branching behavior since variables
+        can have many different types.
+
+        Note: This function assumes the desired node is already in focus and visible to the node inspector
+
+        IMPORTANT NOTE: Only the basic variable types are supported by this function. EntityId not supported at the moment
+            Basic types: "Boolean", "Color", "Number", "String", "Transform", "Vector2", "Vector3", "Vector4"
+
+        Basic type new_value structures
+            Boolean:
+            Color, String: String
+            Number: float
+            Transform: list[math.Vector3, math.Vector3, float]
+            Vectors use corresponding math library vector
+
+        params variable_name: the name of the variable that's selected
+        params variable_type: the type of the variable being configured.
+        params new_value: the new value you want to initialize the variable with
+
+        returns None
+        """
+        #todo
+        #flesh out the rest of the switch
+        #add better verification that the client is trying to modify the correct type of variable
```

Comment:
Remove or backlog.

Reply:
added a GHI (#12972) to finish this part of the file.